cpp/onedal/error_handling.hpp

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include <new>
+#include <system_error>
+#include <stdexcept>
+#include <memory>
+#include <cstring>
+
+namespace dal {
+
+class exception{
+public:
+    virtual const char* what() const noexcept = 0;
+};
+
+class logic_error : public exception {};
+class runtime_error : public exception {};
```

Comment:
Why do you want to make these classes pure virtual?

Reply:
both are abstract classes as we discussed on WG meetings.

Reply:
Can you please clarify why this decision was made?